### DISCUSSION TITLE
Switch from rofi script to drun when typing

### DISCUSSION THREAD

**Discussion by Aya-Komichi** [davatorium/rofi#1918] [Q&A] [answered]
I looked through the documentation, discussions and issues, but did not find any information relevant to this.

Is it possible to make a rofi script so that, when you start typing on the search bar, rofi immediately switches to drun mode? I know this is a weird feature, but it would fit nicely on the script I am making for my own use.

**Comment by DaveDavenport** [comment, marked as answer]
No this is not possible.

**Reply by Aya-Komichi** [reply]
Would you consider having this feature if I open an issue?

**Reply by DaveDavenport** [reply]
Feel free to open a feature request, but there are  things to consider with it.
What if the user has not enabled the drun switcher at run time?  or does not have it compiled in?
A mode  has no access to other modes, atm it can only switch to an index, so even a 'native plugin'  atm cannot do this for all situations.

(also atm. I have little time to work on rofi, so I won't be able to investigate a possible solution soon.)

**Reply by Aya-Komichi** [reply]
> What if the user has not enabled the drun switcher at run time? or does not have it compiled in?

In that case no mode switch could be done, or at least a less expected mode switch would happen. Besides, this possible feature doesn't need to be solely a switch to the drun mode, it could work for other modes as well.

> A mode has no access to other modes, atm it can only switch to an index, so even a 'native plugin' atm cannot do this for all situations.

That sounds reasonable, as long as a mode could switch to another according to their index, that could be serviceable to me (I never found anything about this). Although, I guess the ability to perform an action other than filtering as soon as the user starts writing might still not be there?

> (also atm. I have little time to work on rofi, so I won't be able to investigate a possible solution soon.)

Understandable.

**Reply by DaveDavenport** [reply]
turned this into a request. (#1919)
